### PR TITLE
Feature/new css handles 1x

### PR DESCRIPTION
**Pull request by icazevedo**
# Ta separadinho por commit <3

#### What problem is this solving?
Migrating all components to the new css-handles 1.x. Js components, deprecated or not, didn't have the `classes` prop added to them. The ProductImages component and GradientCollapse should be refactored to ts in another PR.

#### How to test it?

[Workspace](http://icarocomponentshandles--storecomponents.myvtex.com)
[Workspace](http://icarocomponentshandles--eriksbikeshop.myvtex.com)

#### Related to / Depends on

This PR modifies the behavior of the SKUSelector's handles. This PR must only be merged when the changes are addressed in another PR on the `product-summary` repository. This external PR should customize the SKUSelector handles by using the `classes` prop, adding the default `store-components` handles to it. This was previously done by adding a `migrationFrom` inside the SKUSelector, which was kinda hacky.

### CODE REVIEW

File: react/components/BuyButton/Wrapper.js [hunk 1]
```diff
@@ -64,6 +64,9 @@ const BuyButtonMessage = ({ showItemsPrice, skuItems }) => {
   )
 }
 
+/**
+ * @deprecated This component is deprecated. Please use [vtex.add-to-cart-button](https://github.com/vtex-apps/add-to-cart-button) instead.
```
Comment (klzns) [inline react/components/BuyButton/Wrapper.js:68]: You can remove this one, only EnhancedBuyButton comment will be visible

File: react/components/ProductBrand/ProductBrandCssHandles.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/* cssHandlesContext.ts */
```
Comment (klzns) [inline react/components/ProductBrand/ProductBrandCssHandles.ts:1]:
```suggestion
```

File: react/components/SKUSelector/SKUSelectorCssHandles.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/* cssHandlesContext.ts */
```
Comment (klzns) [inline react/components/SKUSelector/SKUSelectorCssHandles.ts:1]:
```suggestion
```

File: react/components/SKUSelector/components/SKUSelector.tsx [hunk 4]
```diff
@@ -366,4 +366,4 @@ const SKUSelector: FC<Props> = ({
   )
 }
 
-export default memo(SKUSelector)
+export default Object.assign(memo(SKUSelector), { cssHandles: CSS_HANDLES })
```
Comment (klzns) [inline react/components/SKUSelector/components/SKUSelector.tsx:369]: I think that is more easy to understand, what do you think?

```suggestion
const MemoizedSKUSelector = memo(SKUSelector)

MemoizedSKUSelector.cssHandles = CSS_HANDLES

export default MemoizedSKUSelector
```

File: react/components/SKUSelector/components/SelectorItem.tsx [hunk 5]
```diff
@@ -166,4 +163,4 @@ const SelectorItem: FC<Props> = ({
   )
 }
 
-export default memo(SelectorItem)
+export default Object.assign(memo(SelectorItem), { cssHandles: CSS_HANDLES })
```
Comment (klzns) [inline react/components/SKUSelector/components/SelectorItem.tsx:166]: Same as above

File: react/components/SearchBar/SearchBarCssHandles.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+/* cssHandlesContext.ts */
```
Comment (klzns) [inline react/components/SearchBar/SearchBarCssHandles.ts:1]:
```suggestion
```

File: react/components/UserAddress/UserAddressCssHandles.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+/* cssHandlesContext.ts */
```
Comment (klzns) [inline react/components/UserAddress/UserAddressCssHandles.ts:1]:
```suggestion
```

**Review by klzns — CHANGES_REQUESTED** [review]

**Review by klzns — CHANGES_REQUESTED** [review]
Also tests are failing, we need to mock `createCssHandlesContext`

image

File: react/components/SKUSelector/components/SKUSelector.tsx (hunk 4 above)
Comment (icazevedo) [reply react/components/SKUSelector/components/SKUSelector.tsx:369]: I agree, but the types of React.memo didn't allow me to do this.

**Review by klzns — APPROVED** [review]
Because of the migrationFrom of the SKU Selector, we should wait for the product-summary PR to be merged before merging this.

**Comment by kaisermann** [comment]
Releated PR: https://github.com/vtex-apps/product-summary/pull/288

**Comment by klzns** [comment]
Workspace with vtex-apps/product-summary#288 linked with this PR
https://summarycss--storecomponents.myvtex.com/apparel---accessories/

File: react/Logo.tsx [hunk 8]
```diff
@@ -2,30 +2,33 @@ import React from 'react'
...
 
 import Placeholder from './components/Logo/LogoPlaceholder'
 import styles from './components/Logo/styles.css'
 
+const CSS_HANDLES = ['logoLink', 'logoImage', 'logoContainer'] as const
+
 interface Props {
   /** Logo's URL */
-  url: string
+  url?: string
```
Comment (victorhmp) [inline react/Logo.tsx:15]: Why should this be optional? I know we do handle that case, so the component doesn't crash. But I think that's more of a protection. I think it makes sense for the `url` to be required

File: react/ProductSKUAttributes.tsx [hunk 9]
```diff
@@ -38,13 +39,15 @@ const getVariationsFromSelectedItem = (
...
 interface Props {
   blockClass?: string
+  /** Used to override default CSS handles */
+  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
 }
 
 /**
  * Name component. Show name and relevant SKU information of the Product Summary
  */
 function ProductSKUAttributes(_props: Props) {
```
Comment (victorhmp) [inline react/ProductSKUAttributes.tsx:49]: Why `_props` instead of `props`? 🤔

**Review by victorhmp — APPROVED** [review]

File: react/ProductSKUAttributes.tsx (hunk 9 above)
Comment (icazevedo) [reply react/ProductSKUAttributes.tsx:49]: Great mysteries of life 😆

**Review by kaisermann — APPROVED** [review]

**Pull request merged**